### PR TITLE
Backports 6.0.5 20210113/v2

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) tickets:

- https://redmine.openinfosecfoundation.org/issues/4896
- https://redmine.openinfosecfoundation.org/issues/4837
- https://redmine.openinfosecfoundation.org/issues/4822

- https://redmine.openinfosecfoundation.org/issues/4818
https://github.com/OISF/suricata/commit/0e70958e67398f599b77ba991dc631e025cae5f4 backported,
https://github.com/OISF/suricata/commit/c1bffa9545b8aa9d0fc64ac6511edd34919135d7 was already in the branch.

- https://redmine.openinfosecfoundation.org/issues/4744
- https://redmine.openinfosecfoundation.org/issues/4467
- https://redmine.openinfosecfoundation.org/issues/4933
- https://redmine.openinfosecfoundation.org/issues/4978
- https://redmine.openinfosecfoundation.org/issues/4971
- https://redmine.openinfosecfoundation.org/issues/4960
- https://redmine.openinfosecfoundation.org/issues/4958
- https://redmine.openinfosecfoundation.org/issues/4954
- https://redmine.openinfosecfoundation.org/issues/4791

suricata-verify-pr: 661

### CODE REVIEW

File: src/app-layer-detect-proto.c [hunk 1]
```diff
@@ -1962,6 +1962,9 @@ int AppLayerProtoDetectConfProtoDetectionEnabledDefault(
         if (node == NULL) {
             SCLogDebug("Entry for %s not found.", param);
             if (default_enabled) {
+                SCLogWarning(SC_ERR_CONF_YAML_ERROR,
+                        "Proto entry not found in suricata.yaml. Enabling anyway."
+                        "Behavior to be discontinued in Suricata 7. Update configuration.");
```
Comment (inashivb) [inline src/app-layer-detect-proto.c:1967]: @victorjulien do you think this is appropriate? Should the message or log level or both be changed?

File: src/app-layer-detect-proto.c [hunk 2]
```diff
@@ -1960,7 +1960,9 @@ int AppLayerProtoDetectConfProtoDetectionEnabled(const char *ipproto,
 
         node = ConfGetNode(param);
         if (node == NULL) {
-            SCLogDebug("Entry for %s not found.", param);
+            SCLogWarning(SC_ERR_CONF_YAML_ERROR,
+                    "Proto entry not found in suricata.yaml. Enabling anyway."
+                    "Behavior to be discontinued in Suricata 7. Update configuration.");
```
Comment (inashivb) [inline src/app-layer-detect-proto.c:1965]: @victorjulien do you think this is appropriate? Should the message or log level or both be changed?

Comment (victorjulien) [reply src/app-layer-detect-proto.c:1965]: How about:

"Config for protocol %s not found, so enabling by default. This behavior will change in Suricata 7, so please update your config. See ticket #4744 for more details."

Where a more thorough explanation of the issue and how to suppress the warning should be in the ticket.

Comment (inashivb) [reply src/app-layer-detect-proto.c:1965]: OK. Thank you!

**Comment by inashivb** [comment]
Replaced w #6808

**Pull request closed without merging**